src/material/list/testing/list-item-harness.ts

```diff
@@ -0,0 +1,248 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessLoader,
+  HarnessPredicate
+} from '@angular/cdk/testing';
+import {
+  MAT_ACTION_LIST_SELECTOR,
+  MAT_LIST_SELECTOR, MAT_NAV_LIST_SELECTOR
+} from '@angular/material/list/testing/list-selectors';
+import {
+  ActionListItemHarnessFilters,
+  BaseListItemHarnessFilters,
+  ListItemHarnessFilters,
+  ListOptionHarnessFilters,
+  NavListItemHarnessFilters,
+  SubheaderHarnessFilters
+} from './list-harness-filters';
+
+function getListItemPredicate
+    <H extends MatListItemHarnessBase, F extends BaseListItemHarnessFilters>
+    (harnessType: ComponentHarnessConstructor<H>, options: F): HarnessPredicate<H> {
+  return new HarnessPredicate(harnessType, options)
+      .addOption('text', options.text,
+          (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text));
+}
+
+/** Harness for interacting with a list subheader. */
+export class MatSubheaderHarness extends ComponentHarness {
+  static hostSelector = '[mat-subheader], [matSubheader]';
```

Comment:
Aren't we running the risk of these selectors going out of sync? Maybe in a follow-up we should pull them out into variables?

Reply:
I think it'll be ok, usually when I remove a selector I do a global search for e.g. 'mat-subheader' to find and fix them all, that should catch this.